docs/user-documentation/moderne-platform/how-to-guides/custom-recipe.md

```diff
@@ -0,0 +1,45 @@
+# Some title
+
+Many people have custom ESLint plugins or other CLI tools that they want to use inside of a recipe so that they can apply it to a large number of repositories.
+
+This guide will walk you through everything you need to know in order to do that.
+
+## Step 1: Create a recipe repository
+
+The first thing you'll need to do is to create a repository for where your recipe should live.
+
+If you are wanting to use a custom ESLint plugin or tool that is available on NPM inside of your recipe, then you should strongly consider forking our [rewrite-codemods repository](https://github.com/moderneinc/rewrite-codemods) as it already includes a lot of the pieces you'll need.
+
+If you are wanting to use some other tool or dependency, consider starting with our [rewrite-recipe-starter repository](https://github.com/moderneinc/rewrite-recipe-starter).
+
+## Step 2: Add your dependencies
+
+If your dependency is available on NPM, please update the `package.json` file inside of your repository and add your dependency:
```

Comment:
What happens if the dependency isn't available on NPM? How do people add dependencies then?

Reply:
ya thats probably a different guide but the answer I think is make a recipe that bundles its depenencies in the jar and has a base recipe that runs the tool in a child process.

Reply:
maybe mention our repo uses a gradle plugin to npm install dependencies as a resource in the jar:
https://github.com/moderneinc/rewrite-codemods/blob/main/build.gradle.kts#L25-L29

Reply:
> maybe mention our repo uses a gradle plugin to npm install dependencies as a resource in the jar: https://github.com/moderneinc/rewrite-codemods/blob/main/build.gradle.kts#L25-L29

Sorry for the noob question but I don't really understand what this is saying. Would you mind explaining what it means to install dependencies as a resource in a jar?